### ISSUE TITLE
Any recommendation for distillation PE？

### ISSUE THREAD

**Issue by Jacobew** [facebookresearch/perception_models#71]
Hi, thanks for your great work on PE!

I’ve tried it on my perception model and got good results.

Now I’m working on distilling a smaller ViT version and wonder:

1. Are there any recommended distillation strategies for small ViT models using only visual data? Would that be less effective than the CLIP-style distillation mentioned in Section C.3?

2. Is there any chance the distillation training code will be released?
 

Appreciate any guidance you can offer! 

@shuminghu @dbolya @PeizeSun @mmaaz60 @Jiankai-Sun

**Comment by dbolya** [comment]
Hi!

1. This depends on how you want to use the model. If your goal is using the result as a CLIP model, then naturally you need to do the CLIP-style distillation. However, for more general distillation check out the losses used for spatial alignment (see appendix, though note that there's a missing minus sign in the current draft of the paper, see #43). Every distillation technique comes with benefits and drawbacks, so you will need to choose one tailored to your downstream use case.

2. There are no plans currently to release training code, but we may release some loss implementations if that is helpful.

**Comment by Jacobew** [comment]
Thanks for your suggestions! They're very helpful.

1. I'm currently working on a 3D perception task that utilizes a PE backbone with several dense decode heads. I'll definitely incorporate the distillation process from the spatial alignment section as you advised.

2. I agree that the loss implementation is crucial, and I'm very much looking forward to seeing it released soon. Your efforts are greatly appreciated!

I still have a few follow-up questions:

- Could you elaborate on the full distillation process for the newly released smaller models (e.g., pe-spatial-tiny/small/base)? Specifically:

    - Are these models distilled solely using the method described in the spatial alignment section?

    - Is the CLIP-style distillation process also involved in their training?

    - Is there any specific initialization strategy used when starting the training of these smaller models?

- Given the variety of distillation methods for Vision Transformers (ViT), I'm particularly interested in the highly promising approach used in [ViT-22B](https://arxiv.org/pdf/2302.05442). I'm curious if your team explored such well-established methods before developing the innovative alignment algorithm presented in your paper. If so, I'd be very grateful if you could share any insights or experiences gained from those trials.

<img width="667" height="167" alt="Image" src="https://github.com/user-attachments/assets/54770077-2425-4634-9882-f5d5d8c6e232" />

@dbolya

**Comment by Jacobew** [comment]
Thank you for your detailed and thoughtful reply—it has truly clarified my questions.

Just to confirm my understanding:

- Is the "direct distillation loss" the cosine similarity loss computed on the spatial features of size H × W?
- And is the "pairwise cosine similarity distillation loss" the MSE loss applied to the flattened HW × HW similarity matrix?
Additionally, have you explored generating the PE spatial series from the PE lang series? In my experiments, the PE lang series outperforms PE core on perception tasks, which is quite interesting.

I would greatly appreciate it if more details on these design choices, along with the exact loss implementations, could be included in the paper or released in the codebase when available.

Looking forward to your insights!

@dbolya
